### PR TITLE
Rev4

### PR DESCRIPTION
**Pull request by maidnl**

### CODE REVIEW

**Review by Rocketct — APPROVED** [review]

File: src/Arduino_MKRIoTCarrier_Relay.cpp [hunk 1]
```diff
@@ -30,22 +30,20 @@ void MKRIoTCarrier_Relay::begin(){
...
+            _pin = 14;
         }
         else if(_index == 2) {
-            _pin = 2;
+            _pin = 13;
         }
     }
     else {
-        
+        /*revisione 2 della scheda con pull up su AREF */
```
Comment (facchinm) [inline src/Arduino_MKRIoTCarrier_Relay.cpp:41]: Please rephrase in english :)

File: src/Arduino_MKRIoTCarrier.cpp [hunk 2]
```diff
@@ -31,22 +31,30 @@ int MKRIoTCarrier::_revision = 0;
 
 int MKRIoTCarrier::begin() {
 
-  pinMode(AREF_PIN,INPUT_PULLDOWN);
```
Comment (facchinm) [inline src/Arduino_MKRIoTCarrier.cpp]: AREF has an external pullup on rev2 while it's hanging on rev1, so the right thing to do is setting as INPUT_PULLDOWN and checking if it's LOW (rev1) or HIGH (rev2)

File: src/Arduino_MKRIoTCarrier.h [hunk 3]
```diff
@@ -59,6 +59,9 @@
 #define TFT_DC        14
 #define TFT_BACKLIGHT 3
 
+#define TFT_CS_OLD        2
```
Comment (facchinm) [inline src/Arduino_MKRIoTCarrier.h:62]: I'd prefer _REV1 instead than _OLD

**Review by facchinm — CHANGES_REQUESTED** [review]

File: src/Arduino_MKRIoTCarrier.h (hunk 3 above)
Comment (Rocketct) [reply src/Arduino_MKRIoTCarrier.h]: done

File: src/Arduino_MKRIoTCarrier.cpp (hunk 2 above)
Comment (Rocketct) [reply src/Arduino_MKRIoTCarrier.cpp]: the logic in last  version is inverted if it's LOW (rev2) or HIGH (rev1)

File: src/Arduino_MKRIoTCarrier_Relay.cpp (hunk 1 above)
Comment (Rocketct) [reply src/Arduino_MKRIoTCarrier_Relay.cpp]: done

**Comment by Rocketct** [comment]
@per1234 shall be added the dependencies to this library ([BME68x Sensor library](https://github.com/BoschSensortec/Bosch-BME68x-Library)) on the github actions, can you handle it?

**Comment by per1234** [comment]
>shall be added the dependencies to this library ([BME68x Sensor library](https://github.com/BoschSensortec/Bosch-BME68x-Library)) on the github actions, can you handle it?

Hi @Rocketct. I have submitted a pull request for it against the head branch of this PR (meaning it will automatically become part of this PR if that one is merged): https://github.com/maidnl/Arduino_MKRIoTCarrier/pull/2

Please let me know if anything else is required.

**Comment by Rocketct** [comment]
sorry @per1234  there was also another library to which the board was dependent Arduino_LSM6DSOX

**Comment by per1234** [comment]
Thanks @Rocketct. Please let me know if I can be of any assistance in the event additional adjustments to the CI are needed.

I did have one question: I see that the "Arduino_LSM6DS3" library dependency has been replaced by "Arduino_LSM6DSOX":

https://github.com/arduino-libraries/Arduino_MKRIoTCarrier/pull/55/files#diff-f267f9ae2f5d5d56a1dbb305e95d40d4408d9ee7f71357f1f67fb18c29b7f082L11

```diff
-depends=Arduino_APDS9960,Arduino_BQ24195,Arduino_HTS221,Arduino_LPS22HB,Arduino_LSM6DS3,Adafruit BusIO,Adafruit DotStar,Adafruit GFX Library,Adafruit ST7735 and ST7789 Library,Arduino_MCHPTouch,TFT_eSPI
+depends=Arduino_APDS9960,Arduino_BQ24195,Arduino_HTS221,Arduino_LPS22HB,Arduino_LSM6DSOX,BME68x Sensor library,Adafruit BusIO,Adafruit DotStar,Adafruit GFX Library,Adafruit ST7735 and ST7789 Library,Arduino_MCHPTouch,TFT_eSPI
```

However, the workflow is still configured to install the "Arduino_LSM6DS3" library:

https://github.com/arduino-libraries/Arduino_MKRIoTCarrier/blob/d0dee5c7c0f3bb2fb0f398edb6b8ef8c516d5f77/.github/workflows/Compile%20Examples.yml#L54

If it is no longer needed, we should remove that library from the workflow configuration for the sake of efficiency and maintainability.

**Comment by facchinm** [comment]
Superseeded by #58

**Pull request closed without merging**